pipeline/environment_function_test.go

```diff
@@ -0,0 +1,132 @@
+package pipeline
+
+import (
+	"errors"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/goreflect/gostructor/infra"
+	"github.com/goreflect/gostructor/tags"
+)
+
+func TestEnvironmentConfig_GetBaseType(t *testing.T) {
+	myStruct := struct {
+		field1 string `cf_env:"testBaseType"`
+	}{}
+	os.Setenv("testBaseType", "tururu")
+	myStruct1 := reflect.Indirect(reflect.ValueOf(myStruct))
+	field1 := myStruct1.Type().Field(0)
+	t.Log("type of field: ", field1.Name+", "+field1.Type.Name())
+	type args struct {
+		context *structContext
+	}
+	tests := []struct {
+		name   string
+		config EnvironmentConfig
+		args   args
+		want   infra.GoStructorValue
+	}{
+		{
+			name:   "get success base type string from environment",
+			config: EnvironmentConfig{},
+			args: args{
+				context: &structContext{
+					StructField: field1,
+					Value:       myStruct1.Field(0),
+				},
+			},
+			want: infra.NewGoStructorTrueValue(reflect.ValueOf("tururu")),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := EnvironmentConfig{}
+			got := config.GetBaseType(tt.args.context)
+			os.Remove("testBaseType")
```

Comment:
Don't do this in a loop. Use defer function right after os.Setenv

Reply:
in our case it is not be a loop it's test contain only one case